File: python/knowledge/AccountabilityType.py
from knowledge.PartyType import PartyType

class AccountabilityType:
    def __init__(self, name: str):
        self.name = name
        self.valid_commissioners = set()
        self.valid_responsibles = set()

    def add_valid_commissioner_type(self, party_type: PartyType):
        self.valid_commissioners.add(party_type)

    def add_valid_responsible_type(self, party_type: PartyType):
        self.valid_responsibles.add(party_type)

    def __repr__(self):
        return f"AccountabilityType({self.name})"

File: src/main/java/org/nc/knowledge/AccountabilityType.java
package org.nc.knowledge;

import java.util.HashSet;
import java.util.Set;

public class AccountabilityType {
    private final String name;
    // Tipi di party validi per questa accountability
    private final Set<PartyType> validCommissionerTypes = new HashSet<>();
    private final Set<PartyType> validResponsibleTypes = new HashSet<>();

    public AccountabilityType(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public Set<PartyType> getCommissioners() {
        return validCommissionerTypes;
    }

    public Set<PartyType> getResponsibles() {
        return validResponsibleTypes;
    }


    public void addValidCommissionerType(PartyType pt) {
        validCommissionerTypes.add(pt);
    }

    public void addValidResponsibleType(PartyType pt) {
        validResponsibleTypes.add(pt);
    }
}